### ISSUE TITLE
Crash when opening the Camera app from the dock

### ISSUE THREAD

**Issue by zeehio** [OpenLauncherTeam/openlauncher#157]
```
Build version: 0.5.1 
Build date: 1980-01-01 00:00:00 
Current date: 2017-06-04 22:43:04 
Device: LGE Nexus 5X 
 
Stack trace:  
android.content.ActivityNotFoundException: No Activity found to handle Intent {  }
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:1809)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1523)
	at android.app.Activity.startActivityForResult(Activity.java:4228)
	at android.app.Activity.startActivityForResult(Activity.java:4186)
	at android.app.Activity.startActivity(Activity.java:4525)
	at android.app.Activity.startActivity(Activity.java:4493)
	at com.benny.openlauncher.widget.AppItemView$Builder$3$1.run(AppItemView.java:224)
	at com.benny.openlauncher.util.Tool$7$1.run(Tool.java:555)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6121)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:889)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)

```

The only openLauncher stuff in the stack trace points to:

https://github.com/OpenLauncherTeam/openlauncher/blob/master/app/src/main/java/com/benny/openlauncher/util/Tool.java#L543

That calls:

https://github.com/OpenLauncherTeam/openlauncher/blob/master/app/src/main/java/com/benny/openlauncher/widget/AppItemView.java#L216

I don't know enough to fix this, but I hope this stack trace helps.

**Comment by BennyKok** [comment]
This seems to be a rare crash as the activity is not found....

**Comment by dkanada** [comment]
@zeehio does this still happen or did it only happen once? Had you recently installed a new camera app or updated your current app? It could be the same issue we are seeing where the AppManager instance isn't always updating the app list correctly.

**Comment by zeehio** [comment]
I removed the camera from the dock by dragging it to the "Remove" area on the top of the screen. For some reason only the "Remove" option appeared (no "Info" or "Uninstall"). When I dropped the "Camera" to the Remove option I saw that the Camera icon was still appearing in the dock. This camera though was working well. This "ghost icon" effect happened to me as well on another app (Instagram) in the desktop.

I had recently installed Instagram, so this may be related to the AppManager instance you mention.

**Comment by MFlisar** [comment]
I've made some similar observation. Normally I see those "empty" apps that only have the remove option if you try to drag & drop them only for automatically created apps on my desktop after I install a new app... Maybe this info helps...

**Comment by zeehio** [comment]
So maybe it is related to this piece of code...

https://github.com/OpenLauncherTeam/openlauncher/blob/cc89dbca98834527d9ee0beb01a3a3c33dfd0bbb/core/src/main/java/com/benny/openlauncher/core/util/ShortcutReceiver.java

I don't know anything about Android development but there is a try/catch that ignores any exception. It does not have any commented reasoning of what exceptions are expected, or why all of them can safely be ignored. I am no expert but it looks a bit wild to me.

**Comment by dkanada** [comment]
That looks like a stub class that was supposed to add shortcut functionality but right now shortcuts are not implemented yet. @MFlisar I almost forgot, it could be that certain apps are updating their activity paths and these values aren't being updated in the database. It could also be an issue with the AppManager class not picking up changes, but my best guess would be the former.

**Comment by MFlisar** [comment]
I've added some logs and I've always seen no component and no activity name in the intent... So changes in the apps is not the reason. Although that's a problem as well but until now I've ignored this problem always as it should be a very rare one

**Comment by BennyKok** [comment]
hum....... I am getting confused

**Comment by MFlisar** [comment]
@BennyKok I must take that back, this may have been wrong. I now tested this and for the automatic shortcut on the desktop it's not true. But it's true what @zeehio wrote, `ShortcutReceiver.java` is the receiver that installs shortcuts for apps on the desktop.

From my point of view, this so called shortcuts are confusing considering the name (mixing them up with shortcuts like in this issue: https://github.com/OpenLauncherTeam/openlauncher/issues/183)

I now solved it this way:
* I try to create an app item if possible and add this one instead of a shortcut item
* now I can remove automatically added items from my desktop after installing a new app
